jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ProjectAccountSettingsManager.kt

```diff
@@ -114,6 +114,14 @@ abstract class ProjectAccountSettingsManager(private val project: Project) : Sim
             recentlyUsedProfiles.add(identifier.id)
 
             selectedCredentialIdentifier = identifier
+
+            identifier.defaultRegionId?.let { regionProvider[it] }?.let { credentialRegion ->
+                selectedRegion.let {
+                    if (it == null || it.partitionId != credentialRegion.partitionId) {
```

Comment:
nit and it's not necessarily better: `it?.partitonId != credentailRegion.partitionId`